specification/network/resource-manager/Microsoft.Network/preview/2022-06-01-preview/networkVerifier.json

```diff
@@ -0,0 +1,1453 @@
+          "type": "string"
+        },
+        "message": {
+          "type": "string"
+        },
+        "target": {
+          "type": "string"
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ErrorResponseBody"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "IPTrafficProperties": {
+      "type": "object",
+      "properties": {
+        "sourceIps": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "destinationIps": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "sourcePorts": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "destinationPorts": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "protocols": {
+          "$ref": "#/definitions/NetworkProtocols"
+        }
+      }
+    },
+    "NetworkProtocols": {
+      "enum": [
+        "TCP",
+        "UDP",
+        "ICMP"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "NetworkProtocols"
+      }
+    },
+    "NetworkVerifier": {
+      "description": "Verifier instance to manage and run analysis.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/NetworkVerifierProperties"
+        },
+        "id": {
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-azure-resource": true,
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ]
+    },
+    "NetworkVerifierList": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NetworkVerifier"
+          }
+        },
+        "nextLink": {
+          "description": "The link used to get the next page of operations.",
+          "type": "string"
+        }
+      }
+    },
+    "NetworkVerifierProperties": {
+      "type": "object",
+      "properties": {
+        "description": {
+          "type": "string"
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState"
+        }
+      }
+    },
+    "NetworkVerifierUpdate": {
+      "type": "object",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "$ref": "#/definitions/NetworkVerifierProperties"
+        }
+      }
+    },
+    "ProvisioningState": {
```

Comment:
>ProvisioningState [](http://example.com/codeflow?start=5&length=17)

Need to add Canceled. Ref: [https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html?q=provisioningstate#rpc005-provisioning-state-semantics-must-be-followed](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html?q=provisioningstate#rpc005-provisioning-state-semantics-must-be-followed).